### PR TITLE
John conroy/fix and test tiles sort dropdown

### PR DESCRIPTION
**Pull request by john-conroy**
Fixes the dropdown ~~so that it doesn't display the selected option in its available options~~ and adds a few tests.

### CODE REVIEW

**Review by mccalluc — CHANGES_REQUESTED** [review]
I would defer to to @tsliaw on this, and I may be thinking about it wrong, but I'm expecting it to work like a pulldown menu, where the items on the list are stable. It's different than a select, where, when you click on it, all the items, including the current one are included.

I might be thinking about it wrong.

**Comment by john-conroy** [comment]
Updated to @tsliaw new design. If this is approved, I will file issues to update the dropdown for the visualization tile selector and also evaluate the component's accessibility.

Screen Shot 2020-10-26 at 3 35 37 PM

Do we want to change the type label for the `display_doi` in the table and tile sort components to 'ID'? I think I would prefer that to switching it for the tiles only. So in this case instead of 'Dataset' it would say 'ID' or maybe 'HuBMAP ID'.

**Comment by mccalluc** [comment]
@john-conroy :
> Do we want to change the type label for the display_doi in the table and tile sort components to 'ID'? I think I would prefer that to switching it for the tiles only. So in this case instead of 'Dataset' it would say 'ID' or maybe 'HuBMAP ID'.

Yeah, that sounds like a good approach to me. Doesn't need to block this, though.

File: context/app/static/js/shared-styles/dropdowns/index.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+const StyledSpan = styled.span`
+  ${(props) => !props.isSelected && `margin-left: calc(${iconSize} + ${props.theme.spacing(1)}px)`};
+`;
+
+function DropdownSelectItem(props) {
+  const { children, isSelected, className, ...rest } = props;
+  return (
+    <FlexMenuItem className={className} {...rest}>
+      {isSelected && <CheckIcon color="primary" />}
+      <StyledSpan isSelected={isSelected}> {children}</StyledSpan>
```
Comment (mccalluc) [inline context/app/static/js/shared-styles/dropdowns/index.js:26]:
```suggestion
      <StyledSpan isSelected={isSelected}>{children}</StyledSpan>
```
?

File: context/app/static/js/shared-styles/dropdowns/index.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  display: flex;
+`;
+
+const CheckIcon = styled(CheckRoundedIcon)`
+  font-size: ${iconSize};
+  margin-right: ${(props) => props.theme.spacing(1)}px;
+`;
+
+const StyledSpan = styled.span`
+  ${(props) => !props.isSelected && `margin-left: calc(${iconSize} + ${props.theme.spacing(1)}px)`};
```
Comment (mccalluc) [inline context/app/static/js/shared-styles/dropdowns/index.js:18]: This is getting a little hairy. Any chance there's a class or something you could use to move the logic into CSS?

**Review by mccalluc — APPROVED** [review]
Thanks! One line in the style is getting complicated, but I don't know if there's an obvious simplification.

File: context/app/static/js/shared-styles/dropdowns/index.js (hunk 2 above)
Comment (john-conroy) [reply context/app/static/js/shared-styles/dropdowns/index.js:18]: Hairy as in it's difficult to read/interpret?

**Review by tsliaw — APPROVED** [review]

**Pull request merged**